panel/src/components/Forms/FormButtons.vue

```diff
@@ -22,29 +58,174 @@
 </template>
 
 <script>
+import dayjs from "dayjs";
+import relativeTime from 'dayjs/plugin/relativeTime';
+dayjs.extend(relativeTime);
+
 export default {
+  data() {
+    return {
+      now: dayjs(),
+      heartbeat: null
+    }
+  },
   computed: {
+    canUnlock() {
+      return this.lock.time + (60 * 2) < this.now.unix();
+    },
     hasChanges() {
       return this.$store.getters["form/hasChanges"](this.id);
     },
+    hasLock() {
+      return this.lock !== null;
+    },
+    hasUnlock() {
+      return this.unlock !== null;
+    },
     id() {
-      return this.$store.state.form.current;
+      return this.$store.getters["form/current"];
+    },
+    isDisabled() {
+      return this.$store.getters["form/isDisabled"];
+    },
+    lock() {
+      return this.$store.getters["form/lock"];
+    },
+    lastLocked() {
+      return this.now.to(dayjs(this.lock.time * 1000));
+    },
+    unlock() {
+      return this.$store.getters["form/unlock"];
     },
-    isLocked() {
-      return this.$store.getters["form/isLocked"];
+    unlockMsg() {
+      return this.canUnlock ? this.$t('unlock') : this.$t('unlock.disabled');
+    },
+    user() {
+      return this.$store.state.user.current.id;
+    }
+  },
+  watch: {
+    hasChanges(current, previous) {
+      // if user started to make changes,
+      // start setting lock on each heartbeat
+      if (current === true && previous == false) {
+        this.onHeartbeat(this.setLock, 30);
```

Comment:
Why is the lock set over and over?

Reply:
Since we base it on the time stamp, we have to keep updating it as long as the user keeps editing.

Reply:
Ah, of course! :)